aicsimageio/readers/array_like_reader.py

```diff
@@ -0,0 +1,370 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import logging
+from typing import List, Optional, Tuple, Union
+
+import dask.array as da
+import numpy as np
+import xarray as xr
+
+from .. import constants, exceptions
+from ..dimensions import DimensionNames
+from ..metadata import utils as metadata_utils
+from ..types import MetaArrayLike
+from .reader import Reader
+
+###############################################################################
+
+logging.basicConfig(
+    level=logging.INFO,
+    format="[%(levelname)4s: %(module)s:%(lineno)4s %(asctime)s] %(message)s",
+)
+log = logging.getLogger(__name__)
+
+###############################################################################
+
+
+class ArrayLikeReader(Reader):
+    """
+    A catch all for numpy, dask, or xarray to Reader interface.
+
+    Parameters
+    ----------
+    image: Union[List[MetaArrayLike], MetaArrayLike]
+        A single, numpy ndarray, dask Array, or xarray DataArray, or list of many.
+        If provided a list, each item in the list will be exposed through the scene API.
+        If provided an xarray DataArray alone or as an element of the list, the
+        known_dims and known_channel_names, kwargs are ignored if there is dimension
+        (or channel coordinate) information attached the xarray object. If the provided
+        xarray object is missing these pieces, the AICSImageIO defaults will be added.
+
+    known_dims: Optional[Union[List[str], str]]
+        A string of known dimensions to be applied to all array(s) or a
+        list of string dimension names to be mapped onto the list of arrays
+        provided to image.
+        Default: None (guess dimensions for single array or multiple arrays)
+
+    known_channel_names: Optional[Union[List[str], List[List[str]]]]
+        A list of string channel names to be applied to all array(s) or a
+        list of lists of string channel names to be mapped onto the list of arrays
+        provided to image.
+        Default: None (create fake channel names for single or multiple arrays)
```

Comment:
What do you mean by `fake channel names`?

Reply:
I guess not fake, but OME default channel ids. `"Channel:0:0"` and such. Will comment as such.